homeassistant/components/directv/__init__.py

```diff
@@ -61,13 +54,12 @@ def get_dtv_data(
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up DirecTV from a config entry."""
     try:
-        dtv_data = await hass.async_add_executor_job(
-            get_dtv_data, hass, entry.data[CONF_HOST]
-        )
-    except RequestException:
+        dtv = DIRECTV(entry.data[CONF_HOST], session=async_get_clientsession(hass))
```

Comment:
Can this raise an exception because it's doing I/O? If it can't raise, let's move it out of the try…except.

Reply:
good point, constructor doesn't do IO. I'll adjust

Reply:
having some test issues with changes similar to IPP . I'm working on fixing them